### PR TITLE
Map some lunar cycle related logic.

### PR DESCRIPTION
**Pull request by i509VCB**
And includes yet another world interface

### CODE REVIEW

File: mappings/net/minecraft/world/LunarWorldView.mapping [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/class_5424 net/minecraft/world/LunarWorldView
+	METHOD method_30271 getDayTime ()J
+	METHOD method_30272 getMoonSize ()F
+		COMMENT Gets the current size of the moon in the world.
+		COMMENT
+		COMMENT @return the size of the moon on a scale of {@code 0.0F} to {@code 1.0F}
+	METHOD method_30273 getLunarIndex ()I
```
Comment (liach) [inline mappings/net/minecraft/world/LunarWorldView.mapping]: I think we've been calling this moon phase before

Comment (haykam821) [reply mappings/net/minecraft/world/LunarWorldView.mapping]: The texture for the moon's various phases is called `moon_phases`, so we should definitely use that.

File: mappings/net/minecraft/world/LunarWorldView.mapping [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+		COMMENT Gets the lunar cycle index of Minecraft's moon.
+		COMMENT
+		COMMENT <p>This is typically used to determine the size of the moon that should be renderered.
```
Comment (haykam821) [inline mappings/net/minecraft/world/LunarWorldView.mapping]:
```suggestion
		COMMENT <p>This is typically used to determine the size of the moon that should be rendered.
```

File: mappings/net/minecraft/world/dimension/DimensionType.mapping [hunk 3]
```diff
@@ -65,6 +66,14 @@ CLASS net/minecraft/class_2874 net/minecraft/world/dimension/DimensionType
 		ARG 0 seed
 	METHOD method_28523 addRegistryDefaults (Lnet/minecraft/class_5455$class_5457;)Lnet/minecraft/class_5455$class_5457;
 		ARG 0 registryManager
+	METHOD method_28528 (J)F
+		ARG 1 time
+	METHOD method_28531 getLunarIndex (J)I
+		COMMENT Gets the lunar cycle index of Minecraft's moon.
+		COMMENT
+		COMMENT <p>This is typically used to determine the size of the moon that should be renderered.
```
Comment (haykam821) [inline mappings/net/minecraft/world/dimension/DimensionType.mapping]:
```suggestion
		COMMENT <p>This is typically used to determine the size of the moon that should be rendered.
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

**Review by liach — APPROVED** [review]

**Review by haykam821 — APPROVED** [review]

File: mappings/net/minecraft/world/dimension/DimensionType.mapping [hunk 4]
```diff
@@ -3,6 +3,7 @@ CLASS net/minecraft/class_2874 net/minecraft/world/dimension/DimensionType
 	FIELD field_20658 biomeAccessType Lnet/minecraft/class_4545;
 	FIELD field_24504 hasCeiling Z
 	FIELD field_24505 ultrawarm Z
+	FIELD field_24752 MOON_SIZE [F
```
Comment (YanisBft) [inline mappings/net/minecraft/world/dimension/DimensionType.mapping]: Why is it not plural?

File: mappings/net/minecraft/world/dimension/DimensionType.mapping [hunk 5]
```diff
@@ -65,6 +66,14 @@ CLASS net/minecraft/class_2874 net/minecraft/world/dimension/DimensionType
 		ARG 0 seed
 	METHOD method_28523 addRegistryDefaults (Lnet/minecraft/class_5455$class_5457;)Lnet/minecraft/class_5455$class_5457;
 		ARG 0 registryManager
+	METHOD method_28528 (J)F
+		ARG 1 time
+	METHOD method_28531 getMoonPhase (J)I
+		COMMENT Gets the lunar cycle index of Minecraft's moon.
```
Comment (YanisBft) [inline mappings/net/minecraft/world/dimension/DimensionType.mapping]:
```suggestion
		COMMENT Gets the moon phase index of Minecraft's moon.
```

**Review by jaskarth — APPROVED** [review]

File: mappings/net/minecraft/world/dimension/DimensionType.mapping (hunk 4 above)
Comment (liach) [reply mappings/net/minecraft/world/dimension/DimensionType.mapping]:
```suggestion
	FIELD field_24752 MOON_SIZES [F
```

**Review by YanisBft — APPROVED** [review]

**Pull request merged**